code/modules/mob/living/carbon/superior_animal/defense.dm

```diff
@@ -0,0 +1,286 @@
+
+				adjustBruteLoss(damage)
+				updatehealth()
+				M.do_attack_animation(src)
+
+/mob/living/carbon/superior_animal/ex_act(severity)
+	..()
+	if(!blinded)
+		if (HUDtech.Find("flash"))
+			flick("flash", HUDtech["flash"])
+
+	var/b_loss = null
+	var/f_loss = null
+	switch (severity)
+		if (1.0)
+			gib()
+			return
+
+		if (2.0)
+			b_loss += 60
+			f_loss += 60
+			ear_damage += 30
+			ear_deaf += 120
+
+		if (3.0)
+			b_loss += 30
+			if (prob(50))
+				Paralyse(1)
+			ear_damage += 15
+			ear_deaf += 60
+
+	adjustBruteLoss(b_loss)
+	adjustFireLoss(f_loss)
+
+	updatehealth()
+
+/mob/living/carbon/superior_animal/handle_regular_status_updates()
+	..()
+	if(status_flags & GODMODE)
+		return
+
+	if(stat == DEAD)
+		blinded = 1
+		silent = 0
+
+	return 1
+
+/mob/living/carbon/superior_animal/adjustBruteLoss(var/amount)
+	. = ..()
+	if (overkill_gib && (amount >= overkill_gib) && (getBruteLoss() >= maxHealth*2))
+		if (bodytemperature > T0C)
+			gib()
+
+/mob/living/carbon/superior_animal/adjustFireLoss(var/amount)
+	. = ..()
+	if (overkill_gib && (amount >= overkill_dust) && (getFireLoss() >= maxHealth*2))
+		dust()
+
+/mob/living/carbon/superior_animal/updatehealth()
+	. = ..() //health = maxHealth - getOxyLoss() - getToxLoss() - getFireLoss() - getBruteLoss() - getCloneLoss() - halloss
+	if (health <= 0)
+		death()
+
+/mob/living/carbon/superior_animal/gib(var/anim = icon_gib, var/do_gibs = 1)
+	if (!anim)
+		anim = 0
+
+	playsound(src.loc, 'sound/effects/splat.ogg', max(10,min(50,maxHealth)), 1)
+	. = ..(anim,do_gibs)
+
+/mob/living/carbon/superior_animal/dust(var/anim = icon_dust, var/remains = dust_remains)
+	if (!anim)
+		anim = 0
+
+	playsound(src.loc, 'sound/effects/Custom_flare.ogg', max(10,min(50,maxHealth)), 1)
+	. = ..(anim,remains)
+
+/mob/living/carbon/superior_animal/death(var/gibbed,var/message = deathmessage)
+	if (stat != DEAD)
+		target_mob = null
+		stance = initial(stance)
+		stop_automated_movement = initial(stop_automated_movement)
+		walk(src, 0)
+
+		density = 0
+		layer = LYING_MOB_LAYER
+
+	. = ..()
+
+/mob/living/carbon/superior_animal/rejuvenate()
+	density = initial(density)
+	layer = initial(layer)
+
+	. = ..()
+
+/mob/living/carbon/superior_animal/pl_effects()
+	. = ..()
+	adjustToxLoss(2)
+
+/mob/living/carbon/superior_animal/get_cold_protection(var/temperature)
+	return cold_protection
+
+/mob/living/carbon/superior_animal/get_heat_protection(var/temperature)
+	return heat_protection
+
+/mob/living/carbon/superior_animal/handle_environment(var/datum/gas_mixture/environment)
+	if(!environment)
+		return
+
+	if (!contaminant_immunity)
+		for(var/g in environment.gas)
+			if(gas_data.flags[g] & XGM_GAS_CONTAMINANT && environment.gas[g] > gas_data.overlay_limit[g] + 1)
+				pl_effects()
+				break
+
+	if(istype(get_turf(src), /turf/space))
+		if (bodytemperature > 1)
+			bodytemperature = max(1,bodytemperature - 10*(1-get_cold_protection(0)))
+
+		if (min_air_pressure > 0)
```

Comment:
ah a thought here

give roaches and spiders a fairly low minimum pressure.
They should still die in a vacuum,  but they should survive just fine in thin atmospheres that would suffocate a human

Reply:
Done